src/lightning_app/utilities/app_helpers.py

```diff
@@ -401,3 +401,116 @@ def default(self, obj: Any) -> Any:
         if callable(getattr(obj, "__json__", None)):
             return obj.__json__()
         return json.JSONEncoder.default(self, obj)
+
+
+def _state_dict(flow: "LightningFlow"):
+    state = {}
+    flows = [flow] + list(flow.flows.values())
+    for f in flows:
+        state[f.name] = f.state
+        extras = f.on_save_state_dict()
+        if extras:
+            state[f.name].update({"extras": extras})
+    for w in flow.works():
```

Comment:
Capturing state variables from work will not ensure that the state the work is exposing is resumable.

Reply:
Yes, this isn't the final code. Right now, this is mainly to ensure the logic works within the test. In the future, we would use the state of the work directly, not the one captured in the flow.

Reply:
Got it